active/swagger-secret-detector.js

```diff
@@ -0,0 +1,282 @@
+// Note that new active scripts will initially be disabled
+// -------------------------------------------------------------------
+// Swagger Secrets & Version Detector - ZAP Active Scan Rule Script
+// -------------------------------------------------------------------
+// Modern ZAP registration using getMetadata() function
+// Import required ZAP Java types for modern registration
+
+var URI = Java.type('org.apache.commons.httpclient.URI');
+var ScanRuleMetadata = Java.type("org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata");
+var CommonAlertTag = Java.type("org.zaproxy.addon.commonlib.CommonAlertTag");
+function getMetadata() {
+return ScanRuleMetadata.fromYaml(`
+id: 100001
```

Comment:
Will need a value reserved in https://github.com/zaproxy/zaproxy/blob/main/docs/scanners.md